pkg/operation/seed/seed.go

```diff
@@ -89,9 +96,119 @@ func List(k8sGardenClient kubernetes.Client, k8sGardenInformers gardeninformers.
 	return seedList, nil
 }
 
+// deployCertificates deploys CA and TLS certificates inside the garden namespace.
+// It takes a map[string]*corev1.Secret object which contains secrets that have already been deployed inside that namespace to avoid duplication errors.
+func deployCertificates(seed *Seed, k8sSeedClient kubernetes.Client, secretsMap map[string]*corev1.Secret) error {
+	var (
+		err                   error
+		data                  map[string][]byte
+		CAPrivateKey          *rsa.PrivateKey
+		CACertificateTemplate *x509.Certificate
+		CACertificatePEM      []byte
+	)
+
+	name := "ca"
+	if val, ok := secretsMap[name]; ok {
+		CAPrivateKey, CACertificateTemplate, CACertificatePEM, err = secretutils.LoadCA(val.Data["ca.key"], val.Data["ca.crt"])
+		if err != nil {
+			return err
+		}
+	} else {
+		CAPrivateKey, CACertificateTemplate, CACertificatePEM, err = secretutils.GenerateCA()
+		if err != nil {
+			return err
+		}
+		data = map[string][]byte{
+			"ca.key": utils.EncodePrivateKey(CAPrivateKey),
+			"ca.crt": CACertificatePEM,
+		}
+		_, err = k8sSeedClient.CreateSecret(common.GardenNamespace, name, corev1.SecretTypeOpaque, data, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	kibanaSecret := generateKibanaTLSSecret(seed, k8sSeedClient)
+
+	if _, ok := secretsMap[kibanaSecret.Name]; !ok {
+		privateKeyPEM, certificatePEM, err := secretutils.GenerateCertificate(kibanaSecret, CACertificateTemplate, CAPrivateKey)
+		if err != nil {
+			return err
+		}
+
+		data = map[string][]byte{
+			"tls.key": privateKeyPEM,
+			"tls.crt": certificatePEM,
+		}
+
+		_, err = k8sSeedClient.CreateSecret(common.GardenNamespace, kibanaSecret.Name, corev1.SecretTypeTLS, data, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// generateIngressBasicAuth generates basic authentication credentials for igress resources.
+// If the seed's kubeconfig contains basic authentication credentials then those will be used,
+// otherwise new ones will be generated and saved in the garden namespace as an opaque secret
+func generateIngressBasicAuth(seed *Seed, k8sSeedClient kubernetes.Client, secretsMap map[string]*corev1.Secret) ([]byte, []byte, error) {
+	var (
+		err               error
+		name              string
+		data              map[string][]byte
+		basicAuthUser     []byte
+		basicAuthPassword []byte
+	)
+
+	config := k8sSeedClient.GetConfig()
```

Comment:
What if we want to use another secret for the ingress basic auth in the future? Can we do that similar to https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/secrets.go#L171?